src/BuildScriptGenerator/DotNetCore/GlobalJsonSdkResolver.cs

```diff
@@ -0,0 +1,371 @@
+﻿// --------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+// --------------------------------------------------------------------------------------------
+
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Extensions.Logging;
+using Microsoft.Oryx.BuildScriptGenerator.Exceptions;
+using Microsoft.Oryx.Common.Extensions;
+
+namespace Microsoft.Oryx.BuildScriptGenerator.DotNetCore
+{
+    public class GlobalJsonSdkResolver
+    {
+        public const string LatestPatch = "latestPatch";
+        public const string Patch = "patch";
+        public const string LatestFeature = "latestFeature";
+        public const string Feature = "feature";
+        public const string LatestMinor = "latestMinor";
+        public const string Minor = "minor";
+        public const string LatestMajor = "latestMajor";
+        public const string Major = "major";
+        public const string Disable = "disable";
+
+        private readonly ILogger<GlobalJsonSdkResolver> _logger;
+
+        public GlobalJsonSdkResolver(ILogger<GlobalJsonSdkResolver> logger)
+        {
+            _logger = logger;
+        }
+
+        public string GetSatisfyingSdkVersion(GlobalJsonModel globalJson, IEnumerable<string> availableSdks)
+        {
+            /*
+                From spec:
+                If no global.json file is found, or global.json doesn't specify an SDK version nor an
+                allowPrerelease value, the highest installed SDK version is used (equivalent to setting rollForward
+                to latestMajor).
+            */
+            if (globalJson?.Sdk == null)
+            {
+                globalJson = new GlobalJsonModel();
+                globalJson.Sdk = new Sdk
+                {
+                    Version = "0.0.000",
+                    RollForward = LatestMajor,
+                };
```

Comment:
I'm confused about how this Sdk object created because I didn't see a constructor within the Sdk class above..